src/Chainweb/Pact/Backend/Types.hs

```diff
@@ -284,43 +310,132 @@ newtype BlockHandler logger p a = BlockHandler
 
         )
 
-newtype PactDbEnv' logger = PactDbEnv' (PactDbEnv (BlockEnv logger SQLiteEnv))
+type ChainwebPactDbEnv logger = PactDbEnv (BlockEnv logger SQLiteEnv)
 
 type ParentHash = BlockHash
 
+-- Context for evaluating Pact transactions, which comes from the parent header.
+-- Note that this isn't actually a parent header, because genesis blocks do not
+-- have parents, but genesis blocks must be evaluated in a context.
```

Comment:
The current wording confuses me a little because of the transition from "which comes from the parent header" to "this isn't actually a parent header.".

Is it still accurate if worded this way?

```
-- | Context for evaluating a Pact transaction.
data ParentContext =
  = BlockParentContext !CWVersion !Cid !BlockHash
  -- ^ The context has a parent, referenced by the given BlockHash.
  | GenesisParentContext !CwVersion !Cid
  -- ^ The context has no parent, the block under evaluation is therefore the genesis block.
```

Reply:
I think this is a legal transformation to make given what I wrote, but that's because what I wrote is bad. Lemme think about this for a bit.

Reply:
Deleted this type entirely in the end.